In [ ]:
import pandas as pd
#importar archivo (datos)
file_path = '/content/sample_data/MetroPT3(AirCompressor).csv'
df = pd.read_csv(file_path)
df = df.drop(columns=["Unnamed: 0"])

#Muestras las primeras 5 filas
print(f"Successfully read '{file_path}'. First 5 rows:")
print(df.head())

Successfully read '/content/sample_data/MetroPT3(AirCompressor).csv'. First 5 rows:
             timestamp    TP2    TP3     H1  DV_pressure  Reservoirs  \
0  2020-02-01 00:00:00 -0.012  9.358  9.340       -0.024       9.358   
1  2020-02-01 00:00:10 -0.014  9.348  9.332       -0.022       9.348   
2  2020-02-01 00:00:19 -0.012  9.338  9.322       -0.022       9.338   
3  2020-02-01 00:00:29 -0.012  9.328  9.312       -0.022       9.328   
4  2020-02-01 00:00:39 -0.012  9.318  9.302       -0.022       9.318   

   Oil_temperature  Motor_current  COMP  DV_eletric  Towers  MPG  LPS  \
0           53.600         0.0400   1.0         0.0     1.0  1.0  0.0   
1           53.675         0.0400   1.0         0.0     1.0  1.0  0.0   
2           53.600         0.0425   1.0         0.0     1.0  1.0  0.0   
3           53.425         0.0400   1.0         0.0     1.0  1.0  0.0   
4           53.475         0.0400   1.0         0.0     1.0  1.0  0.0   

   Pressure_switch  Oil_level  Caudal_impuls

In [ ]:
#Información de las variables
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1516948 entries, 0 to 1516947
Data columns (total 16 columns):
 #   Column           Non-Null Count    Dtype  
---  ------           --------------    -----  
 0   timestamp        1516948 non-null  object 
 1   TP2              1516948 non-null  float64
 2   TP3              1516948 non-null  float64
 3   H1               1516948 non-null  float64
 4   DV_pressure      1516948 non-null  float64
 5   Reservoirs       1516948 non-null  float64
 6   Oil_temperature  1516948 non-null  float64
 7   Motor_current    1516948 non-null  float64
 8   COMP             1516948 non-null  float64
 9   DV_eletric       1516948 non-null  float64
 10  Towers           1516948 non-null  float64
 11  MPG              1516948 non-null  float64
 12  LPS              1516948 non-null  float64
 13  Pressure_switch  1516948 non-null  float64
 14  Oil_level        1516948 non-null  float64
 15  Caudal_impulses  1516948 non-null  float64
dtypes: float64(15), ob

El dataset cuenta con 1.516.948 registros y 16 variables, todas con valores no nulos.
No se identifican datos faltantes, lo que garantiza integridad estructural y permite avanzar directamente hacia el análisis exploratorio y la detección de anomalías sin necesidad de procesos de imputación.

In [ ]:
#Medidas de tendencia central y cuartiles
display(df.describe())

,TP2,TP3,H1,DV_pressure,Reservoirs,Oil_temperature,Motor_current,COMP,DV_eletric,Towers,MPG,LPS,Pressure_switch,Oil_level,Caudal_impulses
count,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06,1.516948e+06
mean,1.367826e+00,8.984611e+00,7.568155e+00,5.595619e-02,8.985233e+00,6.264418e+01,2.050171e+00,8.369568e-01,1.606106e-01,9.198483e-01,8.326640e-01,3.420025e-03,9.914368e-01,9.041556e-01,9.371066e-01
std,3.250930e+00,6.390951e-01,3.333200e+00,3.824015e-01,6.383070e-01,6.516261e+00,2.302053e+00,3.694052e-01,3.671716e-01,2.715280e-01,3.732757e-01,5.838091e-02,9.214078e-02,2.943779e-01,2.427712e-01
min,-3.200000e-02,7.300000e-01,-3.600000e-02,-3.200000e-02,7.120000e-01,1.540000e+01,2.000000e-02,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
25%,-1.400000e-02,8.492000e+00,8.254000e+00,-2.200000e-02,8.494000e+00,5.777500e+01,4.000000e-02,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00
50%,-1.200000e-02,8.960000e+00,8.784000e+00,-2.000000e-02,8.960000e+00,6.270000e+01,4.500000e-02,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00
75%,-1.000000e-02,9.492000e+00,9.374000e+00,-1.800000e-02,9.492000e+00,6.725000e+01,3.807500e+00,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00
max,1.067600e+01,1.030200e+01,1.028800e+01,9.844000e+00,1.030000e+01,8.905000e+01,9.295000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00


El análisis de las medidas de tendencia central y dispersión evidencia que la mayoría de las variables operativas del sistema presentan un comportamiento estable y coherente con los rangos físicos esperados. Las presiones (TP2, TP3, H1 y Reservoirs) se concentran alrededor de valores nominales cercanos a 9 bar, con desviaciones estándar bajas que indican estabilidad en la operación. La temperatura del aceite y la corriente del motor muestran variabilidad moderada, consistente con cambios normales de carga y funcionamiento del compresor. Asimismo, las variables digitales mantienen una distribución binaria adecuada sin valores intermedios, lo que confirma integridad en las señales lógicas. No obstante, se identificó un valor atípico en la variable DV_pressure, cuyo máximo alcanza 9.844 bar, significativamente superior a su media y mediana, lo que sugiere un evento transitorio de alta intensidad que merece análisis específico para determinar si corresponde a una condición operativa normal o a un comportamiento potencialmente anómalo.

In [ ]:
#Investigando un caso sospechoso, pero es real entonces se deja
df[df["DV_pressure"] > 9]

,timestamp,TP2,TP3,H1,DV_pressure,Reservoirs,Oil_temperature,Motor_current,COMP,DV_eletric,Towers,MPG,LPS,Pressure_switch,Oil_level,Caudal_impulses
481277,2020-04-06 14:18:29,10.1,9.798,-0.014,9.844,9.796,71.375,6.07,0.0,1.0,1.0,0.0,0.0,1.0,1.0,1.0


Se identificó un evento atípico en el que DV_pressure alcanza 9.844 bar. Aunque inicialmente podría interpretarse como una anomalía lógica debido a que COMP = 0 y la corriente del motor es aproximadamente 6 A, la definición de la variable COMP indica que el valor 0 corresponde a que el compresor está cargando. Además, el contexto de presiones cercanas a 10 bar y la activación de DV_electric = 1 sugieren que se trata de un evento transitorio de purga del secador y no de un error de sensor. Por tanto, no es un dato corrupto, sino un evento operativo de alta intensidad que podría ser relevante para la detección temprana de fallas.